tuxemon/menu/menu.py

```diff
@@ -495,6 +497,10 @@ def process_event(self, event: PlayerInput) -> Optional[PlayerInput]:
             if event.pressed and self.escape_key_exits:
                 self.close()
 
+            if event.pressed and self.back_state != None and not self.escape_key_exits:
+                self.close()
```

Comment:
is close needed here?  the code for menu will manipulate the state stack by calling `pop_state`.  having to quick stack operations could lead to unexpected behavior.  like popping the current state (the menu), then replacing the one right after it (the start state?)
https://github.com/Tuxemon/Tuxemon/blob/development/tuxemon/menu/menu.py#L673

Reply:
When I remove `self.close()`, it shows this. I couldn't really think of a better solution to this.
![image](https://user-images.githubusercontent.com/35881688/140770864-6417fb39-7d36-44ff-8b1f-11564183893a.png)
`
            if event.pressed and self.back_state is not None:
                self.client.pop_state()
                self.client.replace_state(state_name=self.back_state)
`
Using this block of code has the same solution since it calls pop_state() directly if you want to use this instead.

Reply:
So this is going to be really confusing when i explain this but i will try anyway....
the state stack "remembers" what the previous states were.  in the old code, the WorldState is added to the stack (replaced) before the InputMenu, so closing the input menu goes "back" to the WorldState, by default.  So, the problem becomes..."if i close the input state, how do i also close the world state?".  

the reason why the extra close is required with this implementation is:
* close () will pop the current state (InputMenu)
* the current state is now world state
* then replace the current state (world state), with the start state

```python
        def new_game() -> None:
            # load the starting map
            map_name = prepare.fetch("maps", prepare.CONFIG.starting_map)
            self.client.replace_state("WorldState", map_name=map_name)
            self.client.push_state(
                state_name=InputMenu,
                prompt=T.translate("input_name"),
                callback=set_player_name,
                escape_key_exits=False,
            )
            self.client.push_state(FadeInTransition)
```

Reply:
I think I sort of understand the code a little better. Thanks for the clarification.